tests/src/App/DocumentObserver.cpp

```diff
@@ -0,0 +1,426 @@
+    auto subObjTWithSubObjNormalized {SubObjectT()};
+    // An App::SubObjectT object used to store the result of subObjTWithoutEl.normalized()
+    auto subObjTWithoutElNormalized {SubObjectT()};
+    // An App::SubObjectT object used to store the result of subObjTNoFlatten.normalized()
+    auto subObjTNoFlattenNormalized {SubObjectT()};
+    // An App::SubObjectT object used to store the result of subObjTKeepSubName.normalized()
+    auto subObjTKeepSubNameNormalized {SubObjectT()};
+    // An App::SubObjectT object used to store the result of subObjTConvertIndex.normalized()
+    auto subObjTConvertIndexNormalized {SubObjectT()};
+
+    // Act
+
+    subObjTEmptyNormalized = subObjTEmpty.normalized();
+    subObjTWithoutSubObjNormalized = subObjTWithoutSubObj.normalized();
+    subObjTWithSubObjNormalized = subObjTWithSubObj.normalized();
+    subObjTWithoutElNormalized =
+        subObjTWithoutEl.normalized(SubObjectT::NormalizeOption::NoElement);
+    subObjTNoFlattenNormalized =
+        subObjTNoFlatten.normalized(SubObjectT::NormalizeOption::NoFlatten);
+    subObjTKeepSubNameNormalized =
+        subObjTKeepSubName.normalized(SubObjectT::NormalizeOption::KeepSubName);
+    subObjTConvertIndexNormalized =
+        subObjTConvertIndex.normalized(SubObjectT::NormalizeOption::ConvertIndex);
+
+    // Assert
+
+    // In this case calling SubObjectT::normalized() doesn't have effect as subObjTEmpty has been
+    // initialized with an empty constructor
+    EXPECT_EQ(subObjTEmpty.getSubName(), subObjTEmptyNormalized.getSubName());
+
+    // In this case calling SubObjectT::normalized() doesn't have effect as subObjTWithoutSubObj
+    // hasn't any sub objects
+    EXPECT_EQ(subObjTWithoutSubObj.getSubName(), subObjTWithoutSubObj.getSubName());
+
+    // In this case calling SubObjectT::normalized() changes subObjTWithSubObj subname removing the
+    // "Fusion." part of the of the original subname
+    EXPECT_EQ(subObjTWithSubObjNormalized.getSubName().find(fuseName), std::string::npos);
+
+    // In this case calling SubObjectT::normalized() changes subObjTWithoutEl subname removing the
+    // "Fusion." and "Edge1" parts of the of the original subname
+    EXPECT_EQ(subObjTWithoutElNormalized.getSubName().find(fuseName), std::string::npos);
+    EXPECT_EQ(subObjTWithoutElNormalized.getSubName().find("Edge1"), std::string::npos);
+
+    // In this case calling SubObjectT::normalized() doesn't have effect as neither the
+    // DocumentObject referenced nor the subname of subObjTNoFlatten are changed
+    EXPECT_EQ(subObjTNoFlatten.getSubName(), subObjTNoFlattenNormalized.getSubName());
+
+    // In this case calling SubObjectT::normalized() changes subObjTKeepSubName subname removing the
+    // "Fusion." part of the of the original subname
+    EXPECT_EQ(subObjTKeepSubNameNormalized.getSubName().find(fuseName), std::string::npos);
+
+    // In this case calling SubObjectT::normalized() changes subObjTConvertIndex subname replacing
```

Comment:
Ok, this is a more thorough test that probably does what is meant. I guess, I don't understand what happens here.

Reply:
This test has been reworked adding more cases and related details in my lasy commit.

Hopefully there you'll find answers to some of your questions!